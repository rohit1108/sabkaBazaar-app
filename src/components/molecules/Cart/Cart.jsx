import React, { useContext} from "react";
import { useNavigate } from "react-router-dom";
import CartImage from "../../../static/images/cart.svg";
import { GlobalContext } from "../../../context/GlobalContext";
import { useMediaQuery } from "../../../utils/useMediaQuery";
import Image from "../../atoms/Image/Image";
import Modal from "../../atoms/Modal/Modal";
import CartModal from "../../organisms/CartModal/CartModal";

import "./Cart.scss";

export default function Cart() {
  const {
    cartItems: { count, cartOpen },
    dispatch,
  } = useContext(GlobalContext);

  const countItem = count === 1 ? `${count} item` : `${count} items`;

  const navigate = useNavigate();
  const browserWidth = useMediaQuery("(min-width: 769px)");

  const handleBrowserWidth = () => {
    browserWidth
      ? dispatch({ type: "HANDLE_CART", cartOpen: !cartOpen })
      : navigate("/cartpage");
  };

  const handleOnKeyPress = (e) => {
    e.key === "Enter" && handleBrowserWidth();
  };

  return (
    <>
      <div
        className="cart-wrapper"
        onClick={() => {
          handleBrowserWidth();
        }}
        tabIndex={0}
        onKeyPress={handleOnKeyPress}
      >
        <Image
          source={CartImage}
          className={"cart-wrapper__image"}
          alt={"Cart Image"}
        />
        <p className="cart-wrapper__text" data-testid="count-items">{countItem}</p>
      </div>
      {cartOpen ? (
        <Modal>
          <CartModal cartOpen={cartOpen} />{" "}
        </Modal>
      ) : (
        ""
      )}
    </>
  );
}
