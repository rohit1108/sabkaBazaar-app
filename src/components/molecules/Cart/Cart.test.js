import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
//import { GlobContext } from '../../contexts/AppWideContext';
import Cart from './Cart';
//import { Button } from '../../atoms/CartButton/Button';
import { GlobalContext } from '../../../context/GlobalContext';

describe("Cart Header Component", () => {
    it('renders the component properly and should display the proper initial count in the cart header component', () => {
        render(<GlobalContext.Provider value={{ cartItems: { count: 0 } }}><BrowserRouter><Cart /></BrowserRouter></GlobalContext.Provider>);
        expect(screen.getByTestId("count-items")).toHaveTextContent('0 items');
    });

    it('should have the right cart header component snapshot', () => {
        const tree = renderer.create(<GlobalContext.Provider value={{ cartItems: { count: 0 } }}><BrowserRouter><Cart /></BrowserRouter></GlobalContext.Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    })

});