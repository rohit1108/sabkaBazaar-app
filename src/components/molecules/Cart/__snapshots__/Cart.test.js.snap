// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Header Component should have the right cart header component snapshot 1`] = `
Array [
  <div
    className="cart-wrapper"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={0}
  >
    <img
      alt="Cart Image"
      className="cart-wrapper__image image-wrapper"
      data-testid="image"
      src="cart.svg"
    />
    <p
      className="cart-wrapper__text"
      data-testid="count-items"
    >
      0 items
    </p>
  </div>,
  "",
]
`;
